key_vault_sample_base.js

```diff
@@ -8,17 +8,13 @@
 const dotenv = require("dotenv");
 dotenv.config();
 const util = require('util');
-const msRestAzure = require('ms-rest-azure');
-const ResourceManagementClient = require('azure-arm-resource').ResourceManagementClient;
-const KeyVaultManagementClient = require('azure-arm-keyvault');
-const { DefaultAzureCredential } = require('@azure/identity');
+const { ResourceManagementClient } = require('@azure/arm-resources')
+const { KeyVaultManagementClient } = require("@azure/arm-keyvault")
+const { DefaultAzureCredential, ClientSecretCredential } = require('@azure/identity');
```

Comment:
Please remove the ClientSercretCredential and use DefaultAzureCredential instead.